<?xml version="1.0"?>
<launch>
    <node pkg="path_follower" type="path_follower_node" name="path_follower" output="screen" respawn="true">
        <param name="controller_type" value="dynamic_window" />
    </node>

    <group ns="path_follower">
        <param name="min_velocity" value="0.01" />
        <param name="max_velocity" value="0.7" />
        <param name="waypoint_tolerance" value="2.0" />
        <param name="goal_tolerance" value="0.05" />

        <group ns="controller">
            <!-- parameters, that are dependent of the controller or the robot model -->
            <group ns="dynamic_window">
                <param name="angle_fact" value="0.1" />
                <param name="disobst_fact" value="10.0" />
                <param name="v_fact" value="0.1" />
                <param name="lin_acc" value="2.0" />
                <param name="ang_acc" value="2.0" />
                <param name="lin_dec" value="0.5" />
                <param name="ang_dec" value="0.1" />
                <param name="T_dwa" value="0.26" />
                <param name="v_step" value="0.1" />
                <param name="w_step" value="0.1" />
                <param name="fact_T" value="10.0" />
                <param name="step_T" value="0.13" />
                <param name="obst_dist_thresh" value="0.8" />
                <param name="max_ang_vel" value="0.5" />
                <param name="initial_vel_fact" value="1e-3" />
            </group>
        </group>

        <group ns="collision_avoider">
            <group ns="collision_box">
                <param name="width" value="0.4" />
                <param name="crit_length" value="0.6" />
                <param name="min_length" value="1.0" />
                <param name="max_length" value="1.5" />
                <param name="velocity_factor" value="1.0" />
                <param name="velocity_saturation" value="1.0" />
            </group>
        </group>

        <group ns="supervisor">
            <param name="use_waypoint_timeout" value="false" />
            <group ns="waypoint_timeout">
                <param name="time" value="30.0" />
            </group>

            <param name="use_distance_to_path" value="false" />
            <group ns="distance_to_path">
                <param name="max_dist" value="2.0" />
            </group>

            <param name="use_path_lookout" value="false" />
            <group ns="path_lookout">
                <param name="obstacle_scale_distance" value="1.0" />
                <param name="obstacle_scale_lifetime" value="5.0" />
                <param name="path_width" value="0.5" />
            </group>
        </group>

    </group>

</launch>
